src/Update/Nix/FetchGit.hs

```diff
@@ -99,15 +97,12 @@ fetchTreeToSpanUpdates (FetchNode f) = [revUpdate, sha256Update]
                                   (quoteString (latestSha256 f))
         args = originalInfo f
 
--- Given a Nix expression representing a version value, and the
--- children of the node that contains it, decides whether and how that
--- version string should be updated.  We basically just take the
--- maximum latest commit date of all the fetches in the children.
+-- Given a node of the fetch tree which might contain a version
+-- string, decides whether and how that version string should be
+-- updated.  We basically just take the maximum latest commit date of
+-- all the fetches in the children.
 maybeUpdateVersion :: FetchTree FetchGitLatestInfo -> Maybe SpanUpdate
-maybeUpdateVersion (FetchNode _)    = Nothing
-maybeUpdateVersion (Node Nothing _) = Nothing
-maybeUpdateVersion node@(Node (Just versionExpr) _) =
-  case (fmap latestDate . universeBi) node of
-    [] -> Nothing
-    days -> Just $ SpanUpdate (exprSpan versionExpr)
-                              ((quoteString . pack . show . maximum) days)
+maybeUpdateVersion node@(Node (Just versionExpr) _) = do
+  maxDay <- (maximumMay . fmap latestDate . universeBi) node
```

Comment:
What if I use `toList` (from Data.Foldable) instead of `universeBi`?  Would that be clear enough, since `toList` is not so flexible about what type it returns?